libsvmdata/__init__.py

```diff
@@ -1,5 +1,60 @@
-from .datasets import fetch_libsvm, download_libsvm, NAMES  # noqa
+from .libsvm import DATASETS as libsvm_datasets
 
-supported = list(NAMES.keys())  # noqa
+__version__ = "0.5dev"
+
+ALL_DATABASES = {"LIBSVM": libsvm_datasets}
+ALL_DATASETS = {
+    dataset.dataset_name: dataset
+    for datasets in ALL_DATABASES.values()
+    for dataset in datasets
+}
+SUPPORTED = {
+    database_name: [dataset.name for dataset in datasets]
+    for database_name, datasets in ALL_DATABASES
+}
+
+
+def print_supported_datasets():
+    print("Supported datasets")
+    for database_name, datasets in SUPPORTED.items():
+        print("  - {} :".format(database_name), ", ".join(datasets))
+
+
+def fetch_dataset(dataset_name, replace=False, verbose=False):
```

Comment:
should this go in `core`? I always see `__init__.py`s that are pretty ligh in content

Reply:
Ok it was a bit more difficult than I sought. When importing all the `__init__.py` code into `core.py`, this causes imports clashes : 
- `core.py` depends on `libsvm.py` to import `DATASETS`
- `libsvm.py` depends on `core.py` to import `AbstractDataset`

I fixed that by adding a new file `abstraction.py` where all the stuff related to the abstraction of datasets goes. In `core.py`, I just keep API stuff. Thus, the import relation looks like that : `abstraction.py` <----  `libsvm.py` <--- `core.py`. For further databases, this should lead to more natural imports : `abstraction.py` <----  `any_database.py` <--- `core.py`.

Reply:
Or constants can go into core too, if you want

Reply:
Yes but the same problem with occur if `DATASET` is in `core.py` since this constant depends on `LibsvmDataset` located in `libsvm.py` 😬